src/components/sections/Newsletter/Newsletter.tsx

```diff
@@ -0,0 +1,82 @@
+import type { ComponentPropsWithRef, FormEvent, ReactNode } from 'react'
+import React, { forwardRef, useRef } from 'react'
+import { Form, Label, Input, Button } from '@faststore/ui'
+
+export interface NewsletterProps
+  extends Omit<ComponentPropsWithRef<'form'>, 'onSubmit'> {
+  /**
+   * Heading for the section.
+   */
+  heading: ReactNode
+  /**
+   * Description about the newsletter.
+   */
+  description?: ReactNode
+  /**
+   * Callback function when submitted.
+   */
+  onSubmit: (value: string) => void
+}
+
+const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
+  function Newsletter({ heading, description, onSubmit, ...otherProps }, ref) {
+    const emailInputRef = useRef<HTMLInputElement>(null)
+
+    const handleSubmit = (event: FormEvent) => {
+      event.preventDefault()
+
+      if (emailInputRef.current?.value !== '') {
+        onSubmit(emailInputRef.current!.value)
+      }
+    }
+
+    return (
+      <Form
+        data-store-newsletter
+        ref={ref}
+        onSubmit={handleSubmit}
+        {...otherProps}
+      >
+        <div data-newsletter-content>
```

Comment:
Maybe we can call it `data-newsletter-header`

Reply:
Renamed it [here](https://github.com/vtex-sites/base.store/pull/242/commits/42a382d5889164698d0768ee4ef9c345b8db4475).